cache/disk/disk.go

```diff
@@ -476,7 +476,7 @@ func (c *diskCache) availableOrTryProxy(kind cache.EntryKind, hash string, size
 				}
 
 				if err != nil {
-					log.Printf("Warning: expected item to be on disk, but something happened: %v", err)
+					log.Printf("Warning: expected item to be on disk, but something happened when retrieving %s-%d: %v", hash, size, err)
```

Comment:
Thanks for the contribution.

I'm happy to improve the logging here, but perhaps we can improve this a bit, eg:
* It might be worth logging `blobPath` here instead, like the log in the if case one level up.
* Maybe we could also log something different in each of the cases in the if-else block immediately before this one?

Reply:
Hi! I have adjusted the error as suggested and added some additional variables that should make it easy to figure out which kind of `ReadCloser` is being fetched.
In the end my problem was that I was running the program with too little resources and that caused it to fail in various interesting ways. Once I gave it some more CPUs, it started running like expected. Although this particular error message was not very informative so it would be cool to make it better